base/managers.jl

```diff
@@ -209,6 +209,31 @@ end
 immutable DefaultClusterManager <: ClusterManager
 end
 
+type Sem
```

Comment:
a `NonRentrantSemaphore` (it's best to spell out the name for a type in full) is probably more general useful than just the cluster managers. I think this could go into task.jl with the other parallelism constructs